uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/form/components/leave-type-wizard-form.component.js

```diff
@@ -12,7 +14,165 @@ define([], function () {
     }]
   };
 
-  function LeaveTypeWizardFormController ($log) {
+  function LeaveTypeWizardFormController ($log, sharedSettings) {
     $log.debug('Controller: LeaveTypeWizardFormController');
+
+    var vm = this;
+    var lineBreak = '\n\n';
+
+    vm.leaveTypeCategory = '';
+    vm.leaveTypeCategories = [
+      {
+        value: 'leave',
+        label: 'Leave',
+        icon: 'plane'
+      }
+    ];
+    vm.leaveTypeTitle = '';
+    vm.activeSettingsTabIndex = null;
+    vm.sections = [
+      {
+        name: 'general',
+        label: 'Leave Category'
+      },
+      {
+        name: 'settings',
+        label: 'Leave Category Settings'
+      }
+    ];
+    vm.sectionsTemplatesPath =
+      sharedSettings.sourcePath + 'leave-type-wizard/form/components/form-sections';
+    vm.settingsTabs = [
```

Comment:
This object 
```js
{
  name: 'settings',
  label: 'Leave Category Settings'
}
```
and `vm.settingsTab` are decoupled, even though they are obviously related: you are listing the fields of that section defined in the first object. Doesn't it make sense then to make the fields a property of the section object?

The issue with decoupling is more apparent when you see that you had to write a `getSettingsTabFields` method, hardcoding the reference to that particular section both in the method name and its implementation. What if the section change name? Or you have more sections? I think the code should be reworked to be more flexible

Reply:
They are decoupled because the data structure inside sections are different. The General section has a hardcoded form. The Settings section has a form that would depend on the selected leave type. So I think `getSettingsTabFields` makes sense.

Reply:
the question is why the "General" section has an hardcoded form to begin with. You came up with a data structure and a business logic around it that supports dynamic rendering of forms, i think it should support the content of that section as well

Reply:
It looks too custom to me to render it from the map.

Reply:
But then your solution might be flawed, if it can't support "too" custom content. Isn't the idea of having individual templates for each field there exactly to support whatever each field is (a single text input, a group of checkboxes, some interactive element)?

I know that you haven't implemented the field templates yet, but it seems from your reply that you are not consider these radio buttons a candidate for your system

Reply:
@AkA84 Unfortunately, no. All fields **rows** in the Settings section have the same structure. The fields themselves are of course different, but the **table row** structure is the same for each field. That's why I created an autonomous template for rendering these fields from the map. But in case of the General section (one with the title field) - it's structure is not the same as per fields for Settings section does not match the content, so why not be a bit flexible here?